docs/tutorial.md

```diff
@@ -0,0 +1,721 @@
+        ( _´U`_ )    ╰──────────────────────────╯
+        /___A___\   /
+         |  ~  |
+       __'.___.'__
+     ´   `  |° ´ Y `
+
+    ? Are you ready to build a Blockstack app in React? (Y/n)
+    ```
+
+4. Respond to the prompts to populate the initial app.
+
+    After the process completes successfully, you see a prompt similar to the following:
+
+    ```bash
+    [fsevents] Success:
+    "/Users/theuser/repos/hello-blockstack/node_modules/fsevents/lib/binding/Release/node-v59-darwin-x64/fse.node"
+    is installed via remote npm notice created a lockfile as package-lock.json.
+    You should commit this file. added 1060 packages in 26.901s
+    ```
+
+5. Run the initial application.
+
+    ```bash
+    npm start
+
+    > hello-blockstack@0.0.0 start /Users/moxiegirl/repos/hello-blockstack
+    > webpack-dev-server
+
+    Project is running at http://localhost:8080/
+    webpack output is served from /
+    404s will fallback to /index.html
+    Hash: 4d2312ba236a4b95dc3a
+    Version: webpack 2.7.0
+    Time: 2969ms
+                                     Asset       Size  Chunks                    Chunk Names
+    ....
+      Child html-webpack-plugin for "index.html":
+         chunk    {0} index.html 541 kB [entry] [rendered]
+             [0] ./~/lodash/lodash.js 540 kB {0} [built]
+             [1] ./~/html-webpack-plugin/lib/loader.js!./src/index.html 533 bytes {0} [built]
+             [2] (webpack)/buildin/global.js 509 bytes {0} [built]
+             [3] (webpack)/buildin/module.js 517 bytes {0} [built]
+     webpack: Compiled successfully.
+     ```
+
+   The system opens a browser displaying your running application.
+
+  ![](images/blockstack-signin.png)
+
+   At this point, the browser is running a Blockstack server on your local host.
+   This is for testing your applications only.
+
+6. Choose **Sign in with Blockstack**
+
+   The system displays a prompt allowing you to create a new Blockstack ID or restore an existing one.
+
+    ![](images/create-restore.png)
+
+7. Follow the prompts appropriate to your situation.
+
+    If you are restoring an existing ID, you may see a prompt about your user
+    being nameless, ignore it. At this point you have only a single application
+    on your test server.  So, you should see this single application, with your
+    own `blockstack.id` display name, once you are signed in:
+
+    ![](images/running-app.png)
+
+
+### Add a redirect end point to your application
+
+When a user opens the webapp from the Blockstack browser on an iOS phone,
+you want the web app to redirect the user to your iOS application. The work
+you do here will allow it.
+
+1. From the terminal command line, change directory to the root of your web
+application directory.
+
+2. Use the `touch` command to add a redirect endpoint to your application.
+
+   This endpoint on the web version of your app will redirect iOS users back
+   to your mobile app.
+
+    ```bash
+    $ touch public/redirect.html
```

Comment:
`mkdir public` is required before this command or it shows an error (the public folder doesn't exist).